### ISSUE TITLE
[BUG][Carpet Mod 1.4.147] Wind Charged potion effect behavior changed with optimizedTNT Rule 

### ISSUE THREAD

**Issue by OpecksDev** [gnembon/fabric-carpet#1928]
Wind Charged makes mobs execute an small air burst explosion after death.
When we set the carpet rule "optimizedTNT" to true, the on death effect changes to a normal explosion, including the damage (just like a TNT explosion, even destroying blocks)

-  How to replicate it
Go on creative mode;
Spawn any mob *(only tested with stray) with an egg, and give the potion effect to it (using splash potion or lingering);
Kill it;

This makes the new Ominous Trials unviable.

**Comment by qsser** [comment]
yep i can confirm this happens breeze charges explode like tnt when optimizedTNT is true

**Comment by KeimpeRauwerda** [comment]
It happens with all the wind-charged related effects. I was playing some survival and we kept getting blown up by mobs like Bogged and Strays when they had this effect applied. As soon as a player interacts with a wind charge, it turns into an explosive. So if a Breeze throws a wind charge at you, it works as expected, but when you interact with it by hitting it, it turns very explosive. When killing anything with a maze (from ~2 blocks+) there's also a gigantic explosion and it instantly kills the wielder too.
All our Trial Chambers look like a warzone now.
Turning the rule off fixes it.

**Comment by EmiFree** [comment]
I recorded a video preparing to report this bug, then realized it was already reported. Ill link the video here just in case, I showcased different enchantments with optimizedTNT on and off
[youtube link](https://youtu.be/t9lo0CN0hSw)

**Comment by fishyy119** [comment]
So I tried to fix this bug on the basis of the 1.4.147 version of the code (Inspired by #1975)  in my fork repo (e7260d0001eb6f086b03e30cacdc60aec5d8430d)

I'm really new to fabric (or minecraft mod) development, and the only thing I can guarantee is that the Wind Burst effect will no longer explode like TNT. 
> Maybe my code changed the performance of the original optimizedTNT rules, 

Maybe it can be helpful to somebody.
